swagger/swagger_webservice.go

```diff
@@ -226,10 +226,9 @@ func (sws SwaggerService) composeDeclaration(ws *restful.WebService, pathPrefix
 	pathToRoutes := newOrderedRouteMap()
 	for _, other := range ws.Routes() {
 		if strings.HasPrefix(other.Path, pathPrefix) {
-			if len(other.Path) > len(pathPrefix) && other.Path[len(pathPrefix)] != '/' {
+			if len(pathPrefix) > 1 && len(other.Path) > len(pathPrefix) && other.Path[len(pathPrefix)] != '/' {
```

Comment:
the condition is hard to read and therefore difficult to verify whether all cases are caught.
for this reason, it might be better to define a simple function for it that takes a 2 paths and a prefix and return a bool. 
Then it will be easy to write tests for this and see, by inspecting the samples, that the original issue is solved.